src/profiles_management/create.py

```diff
@@ -0,0 +1,84 @@
+"""Module responsible for creating and updating Profiles based on a PMR.
+
+This module includes both
+1. the high level functions for updating/creating Profiles based on a PM
+2. helpers that will handle the different phases of the above logic
+
+The main function exposed is the create_or_update_profile(pmr), which
+will run an update on the whole cluster's Profiles anad Contributors
+based on a PMR.
+"""
+
+import logging
+from typing import Dict
+
+from charmed_kubeflow_chisme.lightkube.batch import delete_many
+from lightkube import Client
+from lightkube.generic_resource import GenericGlobalResource
+
+from profiles_management.helpers.k8s import get_name
+from profiles_management.helpers.kfam import (
+    list_contributor_authorization_policies,
+    list_contributor_rolebindings,
+)
+from profiles_management.helpers.profiles import list_profiles
+from profiles_management.pmr.classes import ProfilesManagementRepresentation
+
+log = logging.getLogger(__name__)
+client = Client(field_manager="profiles-automator-lightkube")
+
+
+def remove_access_to_stale_profile(profile: GenericGlobalResource):
+    """Remove access to all users from a Profile.
+
+    This is achieved by removing all KFAM RoleBindings and
+    AuthorizationPolicies in the namespace. The RoleBinding / AuthorizationPolicy
+    for the Profile owner will not be touched.
+
+    Args:
+        profile: The lightkube Profile object from which all contributors should be removed.
+    """
+    profile_namespace = get_name(profile)
```

Comment:
`get_name()` will raise and there is no exception handling here, I think we should add something.

Reply:
Hmm for this one, theoretically the exception will never be raised as the `metadata.name` field is mandatory when creating resources 
https://pwittrock.github.io/docs/api-reference/v1.7/#objectmeta-v1-meta:~:text=Is%20required%20when%20creating%20resources